src/middleware.js

```diff
@@ -2,11 +2,37 @@ import path from 'path';
 
 import mime from 'mime-types';
 
+import getETag from 'etag';
+
 import getFilenameFromUrl from './utils/getFilenameFromUrl';
 import handleRangeHeaders from './utils/handleRangeHeaders';
 import ready from './utils/ready';
 
+const { hasOwnProperty } = Object.prototype;
+
 export default function wrapper(context) {
+  const { compiler } = context;
+  const etagRegistry = new Map();
+
+  function computeEtags(stats) {
+    try {
+      const { assets } = stats.compilation;
+      for (const assetId in assets) {
+        if (hasOwnProperty.call(assets, assetId)) {
+          const { existsAt: fsPath } = assets[assetId];
+          const etag = getETag(assets[assetId].source());
+          etagRegistry.set(fsPath, etag);
+        }
+      }
+    } catch (_ignoreError) {} // eslint-disable-line no-empty
+  }
+
+  if (compiler.hooks.done) {
+    compiler.hooks.done.tap('webpack-dev-middleware', computeEtags);
+  } else {
+    compiler.plugin('done', computeEtags);
+  }
```

Comment:
No hooks here

Reply:
We already can load assets here, please look at logic

Reply:
Why you need `iterate`?

Reply:
Anyway if you need this, you have `context`, put this logic inside `done` hooks and collect them `context.etags`

Reply:
Sounds good.

I want to iterate a list of distribution files to efficiently calculate etags. If I only have the memfs instance, I'd have to traverse the entire fs, which isn't optimal.